### PR TITLE
Redirect sometimes

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
We plan to deploy the voting tab first.  For this we will need to be able to switch between svelte and flutter, depending on which tab the user is on.

# Changes

REDIRECT_TO_LEGACY was a binary value, true or false.  We need a third value that lets us see tab changes before they are deployed.  We now have:

Existing string values: "true" "1" "false" "0"
New string values:  "prod", "staging"

# TODO
Update the Notion doc with the environment variable descriptions.

# Tests
I have checked all the following: (on chrome+linux, not that the choice of browser should matter for this)

## Sometimes
This SHOULD use flutter on all tabs except the voting tab.

Deployment: REDIRECT_TO_LEGACY=sometimes -> https://vh6yy-miaaa-aaaaa-aaaya-cai.nnsdapp.dfinity.network

Switching tabs should move back and forth between svelte and flutter.

This should redirect to svelte: https://vh6yy-miaaa-aaaaa-aaaya-cai.nnsdapp.dfinity.network/#/proposals

This should stay on svelte: https://vh6yy-miaaa-aaaaa-aaaya-cai.nnsdapp.dfinity.network/v2/#/proposals

The back button should work across implementations.

## True
This SHOULD use flutter on all tabs 

Deployment: REDIRECT_TO_LEGACY=true -> https://v32cj-3iaaa-aaaaa-aaa2a-cai.nnsdapp.dfinity.network

## False
This SHOULD use svelte everywhere

Deployment: REDIRECT_TO_LEGACY=false -> https://vj4vq-xyaaa-aaaaa-aaaza-cai.nnsdapp.dfinity.network

## Default
This SHOULD use flutter on all tabs, the same as "true"

Deployment: unset REDIRECT_TO_LEGACY -> https://vvypb-ayaaa-aaaaa-aaa3a-cai.nnsdapp.dfinity.network

### CODE REVIEW

File: frontend/svelte/env.config.mjs [hunk 1]
```diff
@@ -55,8 +55,11 @@ const REDIRECT_TO_LEGACY = ["true", "1"].includes(
 )
   ? true
   : ["false", "0"].includes(process.env.REDIRECT_TO_LEGACY) ||
-    ENVIRONMENT === "local"
+    ENVIRONMENT === "local" // Note: This is also deployed to testnets.
```
Comment (lmuntanerDfinity) [inline frontend/svelte/env.config.mjs:58]: Can you update the comment above the definition as well?

**Review by lmuntanerDfinity — APPROVED** [review]
Minor comment to improve comment documentation.

Looking forward to getting rid of Flutter so that we can also remove this complex logic, hehe.

File: frontend/svelte/env.config.mjs (hunk 1 above)
Comment (bitdivine) [reply frontend/svelte/env.config.mjs:58]: Done.

Comment (lmuntanerDfinity) [reply frontend/svelte/env.config.mjs:58]: thanks!

File: frontend/svelte/env.config.mjs [hunk 2]
```diff
@@ -50,14 +50,16 @@ const OWN_CANISTER_URL = `https://${OWN_CANISTER_ID}${domain}/`;
 // - true while flutter is being replaced by svelte
 // - false after flutter has been replaced, but before all scaffolding has been removed
 // - the flag may then be removed.
-const REDIRECT_TO_LEGACY = ["true", "1"].includes(
+const REDIRECT_TO_LEGACY = ["true", "1", "prod"].includes(
```
Comment (lmuntanerDfinity) [inline frontend/svelte/env.config.mjs:53]: question maybe for another PR, do we need to still rely on "true", "false", "1" and "0", can't we clean and change where we use it?

**Review by lmuntanerDfinity — APPROVED** [review]
comment for the future, I like it, much more readable!

**Pull request merged**